docs/src/proposals/off-chain-message-signing.md

```diff
@@ -0,0 +1,78 @@
+# Off-chain message signing
+## Motivation
+
+There is ecosystem demand for a method of signing non-transaction messages with
+a Solana wallet. Typically this is some kind of "proof of wallet ownership" for
+entry into a whitelisted system.
+
+Some inspiration can be gleaned from relevant portions of Ethereum's
+[EIP-712](https://eips.ethereum.org/EIPS/eip-712)
+
+## Considerations
+
+* Security
+  * Off-chain message signatures should not be valid transaction message signatures
+* Future-proofing
+  * Versioning
+  * Co-exist with versioned transaction messages and extended length transactions
+* Compatibility
+  * Hardware wallet signing
+  * Localization
+
+## Payload format
+
+| 0x00  ..= 0x0f | 0x10           | 0x11           | 0x12  ..= 0x13 | 0x14 .. |
+| :------------: | :------------: | :------------: | :------------: | :-----: |
+| Signing Domain | Header Version | Message Format | Message Length | Message |
+
+The payload consists of a 16-byte signing domain specifier, 4-byte header and
+n-byte message.
+
+### Signing Domain Specifier
+The signing domain specifier is a prefix byte string used to give similar structure
+to all off-chain message signatures. We assign its value to be:
+```
+b"\xffsolana offchain"
+```
+The first byte, `\xff`, was chosen for the following reasons
+1. It corresponds to a value that is illegal as the first byte in a transaction
+`MessageHeader`.
+1. It avoids unintentional misuse in languages with C-like, null-terminated strings.
+
+The remaining bytes, `b"solana offchain"`, were chosen to be descriptive and
+reasonably long, but are otherwise arbitrary.
```

Comment:
What about something generic like `\xffsigned message\xff`?

This format being reusable / multi-chain can help wallets simplify their implementation if the chain they are supporting doesn't already define this.

Reply:
that `\xff` is totally solana-specific.  ethereum uses `\x19` for the same purpose.  that is, "shit, what can we use to discriminate this thing that we didn't think about 'til way too late."  I don't think this is a constraint we should force on ourselves

Reply:
Similar situation with the QUIC node server cert. It still uses the root signing domain (no prefix)

Reply:
\x19 was chosen because it is the length of the string "Ethereum Signed Message:\n", which came from Bitcoin's "\x18Bitcoin Signed Message:\n". It turned out later that \x19 was a good choice because RLP encoded transactions can't start with that value. And it was selected for a signed data standard: https://eips.ethereum.org/EIPS/eip-191

Reply:
I propose adding a disclaimer such as: "Do not reuse the signing key for any applications except Solana off-chain and transaction signing." 
Multi-chain functionality can be implemented through mutual authentication (i.e. Solana off-chain signature on Other-Chain key and vice versa); or through BIP32-Ed25519 extended public keys/HD wallets.

Reply:
> I propose adding a disclaimer such as: "Do not reuse the signing key for any applications except Solana off-chain and transaction signing." Multi-chain functionality can be implemented through mutual authentication (i.e. Solana off-chain signature on Other-Chain key and vice versa); or through BIP32-Ed25519 extended public keys/HD wallets.

not sure that'll fly. afaik the clowns brigading the off-chain message signing for ledger app issue all want to authenticate their nft ownership